projects/igniteui-angular/src/lib/services/overlay/overlay.ts

```diff
@@ -684,6 +714,8 @@ export class IgxOverlayService implements OnDestroy {
                 this._overlayElement = null;
             }
             this.removeCloseOnEscapeListener();
+        } else if (!this._overlayInfos[this._overlayInfos.length - 1].settings.closeOnEscape) {
```

Comment:
if there are two overlays open, e.g. dialog in another dialog, and if the first one has `closeOnEscape` set to `true`, when the second one is closed this will remove the handler of the first one, right?
This does not look ok.

Reply:
When the overlay, e.g. the dialog is closed in the `detach()` method, only the click listener is removed `this.removeOutsideClickListener(info);`.

Then the `cleanUp()` method is called to clear everything else. In this method, when checking `if (this._overlayInfos.length === 0) {}` or that is, only when there is an overlay, e.g. a dialog, then the escape listener is also removed `this.removeCloseOnEscapeListener();`. Removing the escape listener only happens here.

When another dialog or nested dialog is open, if the first one has `closeOnEscape` set to `false` and the last open dialog is closed, then the escape listener will never be removed - we will not enter the check because we have another open dialog, and this dialog has set `closeOnEscape` to `false` and there will still be an escape listener.

This check checks, after closing and detaching the last open overlay, e.g. dialog, whether the one before it has the `closeOnEscape ` property set to `false`, and if so, the escape listener is removed.

Regarding the question, if there are two overlays open, e.g. dialogs, and if the first one has `closeOnEscape ` set to `true`, when the second one is closed this will not remove the handler of the first one.